src/core/lbgpu_cuda.cu

```diff
@@ -4275,22 +4275,32 @@ void lb_lbfluid_get_population( int xyz[3], float population_host[LBQ], int c )
  * @param *particle_data        Pointer to the particle position and velocity (Input)
  * @param *u_gpu                Pointer to float array (Output)
  * @param node_f                Struct for node force (Input)
+ * @param coupling              int denoting wether to use two- (0) or three-point (1) coupling for the velocity (1) interpolation (Input)
 */
-__global__ void lb_lbfluid_get_fluid_velocity_at_particle_positions_kernel(LB_nodes_gpu n_a, CUDA_particle_data *particle_data, float* u_gpu, LB_node_force_gpu node_f){
+__global__ void lb_lbfluid_get_fluid_velocity_at_particle_positions_kernel(LB_nodes_gpu n_a, CUDA_particle_data *particle_data, float* u_gpu, LB_node_force_gpu node_f, int coupling){
   unsigned int index = blockIdx.y * gridDim.x * blockDim.x + blockDim.x * blockIdx.x + threadIdx.x;
   if (index < para.number_of_particles) {
     float position[3];
+    unsigned int node_index[8];
+    float mode[19*LB_COMPONENTS];
+    float delta[8];
     position[0] = particle_data[index].p[0];
     position[1] = particle_data[index].p[1];
     position[2] = particle_data[index].p[2];
     // Do the velocity interpolation
-    get_interpolated_velocity(n_a, position, &u_gpu[3*index], node_f);
+#ifndef SHANCHEN //because SHANCHEN uses dv in the two point coupling and we only give NULL pointer.
+    if (coupling == 0) {
+        interpolation_two_point_coupling(n_a, position, node_index, mode, nullptr, delta, &u_gpu[3*index]);
+    } else if (coupling == 1) {
+        interpolation_three_point_coupling(n_a, position, node_index, nullptr, delta, &u_gpu[3*index]);
+    }
+#endif
   }
 }
 
 /** interface to get fluid velocities at particle positions
 */
-std::vector<float> lb_lbfluid_get_fluid_velocity_at_particle_positions() {
+std::vector<float> lb_lbfluid_get_fluid_velocity_at_particle_positions(std::string coupling="twopoint") {
```

Comment:
Please be consistent with the rest of LB and use the strings `"2pt"` and `"3pt"`.